src/fundus/parser/utility.py

```diff
@@ -27,15 +26,32 @@
 from fundus.parser.data import ArticleBody, ArticleSection, TextSequence
 
 
+def normalize_whitespace(text: str) -> str:
+    return " ".join(text.split())
+
+
 @total_ordering
 @dataclass(eq=False)
 class Node:
     position: int
     node: lxml.html.HtmlElement = field(compare=False)
     _break_selector: ClassVar[XPath] = XPath("*//br")
 
-    def striped(self, chars: Optional[str] = None) -> str:
-        return str(self).strip(chars)
+    # one could replace this recursion with XPath using an expression like this:
+    # //*[not(self::script) and text()]/text(), but for whatever reason, that's actually 50-150% slower
+    # than simply using the implemented mixture below
+    def text_content(self, excluded_tags: Optional[List[str]] = None) -> str:
```

Comment:
Just in general to understand the problem at hand: 

> This is due to Fundus using a broad XPath expression string() to extract text from nodes.

I don't see an XPATH that uses the `string()` expression anywhere. From the previous code, there is only the `self._get_break_preserved_node().text_content()`. So does lxml use the `string()` XPATH to extract the text content internally? At least in the text content attribute in the HTML inspector for the FOCUS test article, I could see the javascript there: 
```
textContent: 'Über unsere Gutschein-Seite erhalten Sie exklusiv noch ein zweites Geschenk, wenn Sie den dort abrufbaren Code ab 150 Euro Bestellwert im Warenkorb hinterlegen.  !function(){var t=window.addEventListener?"addEventListener":"attachEvent";(0,window[t])("attachEvent"==t?"onmessage":"message",function(t){if("string"==typeof t.data&&t.data.indexOf("documentHeight:")>-1){var e=t.data.split("documentHeight:")[1],n=(e=t.data.split("documentHeight:")[1],e=parseInt(e)+30,function(t){for(var e=document.getElementsByTagName("iframe"),n=0,a=e.length;n<a;n++)if(e[n].contentWindow===t.source&&-1!=e[n].id.indexOf("chipwidget_"))return e[n];return null}(t));n&&(n.style.height=e+"px")}},!1)}();'
```

Reply:
Yes, `lxml.html.HTMLElement.text_content()` is just a wrapper for `HTMLElement.xpath("string()")`.